vertical-pod-autoscaler/pkg/admission-controller/logic/recommendation_provider.go

```diff
@@ -61,17 +64,36 @@ func NewRecommendationProvider(vpaLister vpa_lister.VerticalPodAutoscalerLister,
 	}
 }
 
-func limitMatchesRequest(limit, request *resource.Quantity) bool {
-	// Limit not set, it cannot match the request.
-	if limit == nil || limit.Value() == 0 {
-		return false
+func limitMatchesRequest(originalLimit, originalRequest, recommendedRequest *resource.Quantity) *resource.Quantity {
+	// originalLimit not set, don't set limit.
+	if originalLimit == nil || originalLimit.Value() == 0 {
+		return nil
+	}
+	// originalLimit set but originalRequest not set - K8s will treat the pod as if they were equal,
+	// recommend limit equal to request
+	if originalRequest == nil || originalRequest.Value() == 0 {
+		return recommendedRequest
 	}
-	// Limit set but request not set - K8s will treat the pod as if they were equal.
-	if request == nil || request.Value() == 0 {
-		return true
+	// originalLimit and originalRequest are set. If they are equal recommend limit equal to request.
+	if originalRequest.MilliValue() == originalLimit.MilliValue() {
+		return recommendedRequest
 	}
-	// Limit and request are set. Check if they are equal.
-	return request.Value() == limit.Value()
+
+	// Input and output milli values should fit in int64 but intermediate values might be bigger.
+	originalMilliRequest := big.NewInt(originalRequest.MilliValue())
```

Comment:
Does it make sense to take MilliValue for memory? I think the Value() is in bytes

Reply:
I don't think it hurts to use millibytes for memory. It limits max value this code can work with to ~2^54 bytes, orders of  magnitude more than we might need to use those years.

In the meantime it simplifies code somewhat.